docs/src/literate/ns_vs_diffeq.jl

```diff
@@ -0,0 +1,463 @@
+#
+# ```math
+#  \begin{aligned}
+#    \partial_t v &= \nu \Delta v - (v \cdot \nabla) v - \nabla p \\
+#               0 &= \nabla \cdot v
+#  \end{aligned}
+# ```
+#
+# where $v$ is the unknown velocity field, $p$ the unknown pressure field
+# and $\nu$ the dynamic viscosity. We assume a constant density of 1 for the fluid
+# and negligible coupling between the velocity components.
+# Finally we see that the pressure term appears only in combination with the gradient
+# operator, so for any solution p the function p + c is also an admissible solution, if
+# we do not impose Dirichlet conditions on the pressure. To resolve this we introduce the
+# implicit constraint that $ \int_\Omega p = 0 $.
+#
+# Our setup is derived from [Turek's DFG benchmark](http://www.mathematik.tu-dortmund.de/~featflow/en/benchmarks/cfdbenchmarking/flow/dfg_benchmark1_re20.html).
+# We model a channel with size $0.41 \times 2.2$ and a hole of radius $0.05$ centered at $(0.2, 0.2)$.
+# The left side has a parabolic inflow profile, which is ramped up over time, modeled as the Dirichlet condition
+# $v_x(t) \mapsto 4*v_{in}(t)*y*(0.41-y)/0.41^2$ where $v_{in}(t) = clamp(t, 0.0, 1.0)$. With a viscosity of $\nu = 0.001$
+# this is enough to induce turbulence behind the cylinder which leads to vortex shedding. The top and bottom of our
+# channel have no-slip conditions, i.e. v = (0,0), while the right boundary has the do-nothing boundary condtion
+# $\nu \partial_n v - p n = 0$ to model outflow. With these boundary conditions we can choose the zero solution as a
+# feasible initial condition.
+#
+# ### Semi-Discrete Weak Form
+#
+# ```math
+#  \begin{aligned}
+#    \int \partial_t v \cdot \varphi &= - \int \nu \nabla v : \nabla \varphi - \int (v \cdot \nabla) v \cdot \varphi + \int p \nabla \cdot \varphi + \int_{\partial \Omega_{N}} (\nu \partial_n v - p n ) \cdot \varphi \\
+#                                  0 &= \int \nabla \cdot v \psi
+#  \end{aligned}
+# ```
+# where $\varphi$ and $\psi$ are suitable test functions.
+#
+# Now we can discretize the problem as usual with the finite element method
+# utilizing Taylor-Hood elements (Q2Q1) to yield a stable discretization.
+# ```math
+#  M [\hat{\mathbf{v}}, \hat{p}] = K [\hat{\mathbf{v}}, \hat{p}] + [N(\hat{\mathbf{v}}, \hat{\mathbf{v}}, \hat{\varphi}), 0]
+# ```
+# Here M is the singular block mass matrix, K is the discretized Stokes operator and N the non-linear advective term.
+#
+#
+# ## Commented Program
+#
+# Now we solve the problem in Ferrite with DifferentialEquations.jl. What follows is a program spliced with comments.
+# The full program, without comments, can be found in the next [section](@ref ns_vs_diffeq-plain-program).
+#
+# First we load Ferrite, and some other packages we need
+using Ferrite, SparseArrays, BlockArrays, LinearAlgebra, UnPack
+# Since we do note need the complete DifferentialEquations suite just load the required ODE infrastructure, which can also handle
```

Comment:
```suggestion
# Since we do not need the complete DifferentialEquations suite, we just load the required ODE infrastructure, which can also handle
```